contrib/vcloud/podman_containerized_tool.py

```diff
@@ -0,0 +1,187 @@
+        "--workdir",
+        "/mnt",
+    ]
+
+    # Create a container that does nothing but keeps running
+    command = (
+        ["podman", "create", "--entrypoint", '[""]', "--rm"]
+        + volumes
+        + [image, "/bin/sh"]
+    )
+
+    logging.debug(
+        "Command to start container: %s",
+        shlex.join(map(str, command)),
+    )
+    res = subprocess.run(
+        command,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.DEVNULL,
+        stdin=subprocess.DEVNULL,
+        check=True,
+    )
+    container_id = res.stdout.decode().strip()
+
+    subprocess.run(
+        ["podman", "init", container_id],
+        check=True,
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+        stdin=subprocess.DEVNULL,
+    )
+
+    container_pid = (
+        subprocess.run(
+            ["podman", "inspect", "--format", "{{.State.Pid}}", container_id],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.DEVNULL,
+            stdin=subprocess.DEVNULL,
+        )
+        .stdout.decode()
+        .strip()
+    )
+
+    def join_ns(namespace):
+        namespace = f"/proc/{container_pid}/ns/{namespace}"
+        logging.debug("Joining namespace %s .", namespace)
+        with open(namespace, "rb") as f:
+            libc.setns(f.fileno(), 0)
+
+    try:
+        logging.debug("Joining namespaces of container %s.", container_id)
+
+        necessary_namespaces = frozenset(("user", "mnt"))
+
+        # The user namespace must be joined first
+        # because the other namespaces depend on it
+        join_ns("user")
+
+        for namespace in os.listdir(f"/proc/{container_pid}/ns"):
+            if namespace in necessary_namespaces:
+                continue
+            try:
+                # We try to mount all listed namespaces, but some might not be available
+                join_ns(namespace)
+
+            except OSError as e:
+                logging.debug(
+                    "Failed to join namespace %s: %s", namespace, os.strerror(e.errno)
+                )
+
+        # The mount namespace must be joined so we want
+        # to fail if we cannot join the mount namespace.
+        # mnt must be joined last because after joining it,
+        # we can no longer access /proc/<container_pid>/ns
+        join_ns("mnt")
+
+        os.chdir(TOOL_DIRECTORY_MOUNT_POINT)
+        return container_id
+
+    except OSError as e:
+        raise BenchExecException(
+            "Joining the podman container failed: " + os.strerror(e.errno)
+        )
+
+
+@tooladapter.CURRENT_BASETOOL.register  # mark as instance of CURRENT_BASETOOL
+class PodmanContainerizedTool(ContainerizedTool):
+    tool_directory: str
+    image: str
+    container_id: Optional[str]
+
+    def __init__(self, tool_module, config, image):
+        assert (
+            config.tool_directory
+        ), "Tool directory must be set when using podman for tool info module."
```

Comment:
This is currently the only place where this is checked and thus causes a crash of `vcloud-benchmark.py`. That a tool directory is given should be checked higher up (where other config options are also handled) and with a proper error message for the user instead of a crash.

Reply:
Hi, I now check for this in the `hook_load_tool_info` function.